charts/internal/falco/values.yaml

```diff
@@ -92,13 +92,13 @@ resources:
   # to #falco slack channel for more info about it.
   requests:
     cpu: 100m
-    memory: 1024Mi
+    memory: 512Mi
   # -- Maximum amount of resources that Falco container could get.
   # If you are enabling more than one source in falco, than consider to increase
   # the cpu limits.
   limits:
     # cpu: 1000m
-    memory: 2048Mi
+    memory: 1024Mi
```

Comment:
You could have keep the limits higher, they do not affect scheduling decisions as are requests doing. 
Generally speaking, you may find some insights in https://gardener.cloud/docs/gardener/autoscaling/shoot_pod_autoscaling_best_practices/ and implement better resource allocation for the falco daemon set